lib/semantics/tools.h

```diff
@@ -248,6 +248,19 @@ template<typename T> std::optional<std::int64_t> GetIntValue(const T &x) {
   }
 }
 
+template<typename T> bool IsZero(const T &expr) {
+  const std::optional<std::int64_t> value{GetIntValue(expr)};
+  return value && *value == 0;
+}
+
+template<typename T>
+static void SayIfZero(T &expr, SemanticsContext &context,
```

Comment:
Why is this template function `static` and the preceding one not?

Why not `const T &expr`?

Reply:
No good reason.  I'll change them.